In [1]:
import astropy.io.fits as fits
import numpy as np

Open up the quasar catalog using astropy.io.fits, and in particular look at the first HDU data and headers.

In [2]:
cat = fits.open('/home/ray/Downloads/DR14Q_v4_4.fits')
hdu1 = cat[1]

headers = hdu1.header
data = hdu1.data

Let's look at all the headers in the first HDU.

In [34]:
print(list(headers.values()))

['BINTABLE', 8, 2, 1505, 526356, 0, 1, 105, 'DR14Q_v4_4', 'SDSS_NAME', '18A', '-', 'RA', 'D', 'deg', 'DEC', 'D', 'deg', 'THING_ID', 'J', '-', 'PLATE', 'I', '-', 'MJD', 'J', '-', 'FIBERID', 'I', '-', 'SPECTRO', '4A', '-', 'Z', 'D', '-', 'Z_ERR', 'D', '-', 'SOURCE_Z', '4A', '-', 'Z_VI', 'D', '-', 'Z_PIPE', 'D', '-', 'Z_PIPE_ERR', 'D', '-', 'ZWARNING', 'I', '-', 'Z_PCA', 'D', '-', 'Z_PCA_ER', 'D', '-', 'Z_MGII', 'D', '-', 'BOSS_TARGET1', 'K', '-', 'ANCILLARY_TARGET1', 'K', '-', 'ANCILLARY_TARGET2', 'K', '-', 'EBOSS_TARGET0', 'K', '-', 'EBOSS_TARGET1', 'K', '-', 'EBOSS_TARGET2', 'K', '-', 'N_SPEC_SDSS', 'I', '-', 'N_SPEC_BOSS', 'I', '-', 'N_SPEC', 'I', '-', 'PLATE_DUPLICATE', '51J', '-', 'MJD_DUPLICATE', '51J', '-', 'FIBERID_DUPLICATE', '51J', '-', 'SPECTRO_DUPLICATE', '51A', '-', '(1,51)', 'BI_CIV', 'D', 'km/s', 'ERR_BI_CIV', 'D', 'km/s', 'RUN_NUMBER', 'I', '-', 'RERUN_NUMBER', 'A', '-', 'COL_NUMBER', 'I', '-', 'FIELD_NUMBER', 'I', '-', 'OBJ_ID', 'A', '-', 'PSFFLUX', '5D', 'nanomaggies', 

There are lots of headers in the catalog, reference the catalog data model for descriptions of each header. For now, let's pull out all of the redshifts from the catalog.


In [38]:
redshift = data['z']
print(len(redshift))

526356


We have 526356 quasars in the catalog, and 'redshift' is an array of the redshifts for each one. If we want to select a subsample of the catalog based on redshift, we could do:


In [46]:
ix = (redshift > 4)&(redshift < 5)
cut_data = data[ix]

print(len(cut_data))

2056


We introduced cuts to the catalog such that quasars must be between redshifts 4 and 5. This reduced the sample down to 2056 quasars. If we want to access other properties of the catalog after the cuts, we should do operations on the new catalog. Printing right ascensions:

In [51]:
print(cut_data['RA'])


[1.54184378e-01 1.94549918e-01 2.99565151e-01 ... 3.59493964e+02
 3.59652226e+02 3.59748885e+02]


In [ ]:
for i, x in enumerate(BAL_Indicator)
    condition = i
    if x > 0:
        failure = x
        null_list = np.concatenate([null_list, [failure]])
    else:
        success = condition
        data_list = np.concatenate([data_list, [success]])